linter.lua

```diff
@@ -145,14 +246,27 @@ local function update_cache(doc)
   if not lints[1] then return end
 
   local d = {}
-  local path = system.absolute_path(doc.filename)
-  for _, l in ipairs(lints) do
-    get_file_warnings(d, path, l)
+  local function on_warning()
```

Comment:
This function is not used anywhere. If possible run luackeck to check this cases.